src/com/tutorial/main/Handler.java

```diff
@@ -1,51 +0,0 @@
-package com.tutorial.main;
-
-import java.awt.Graphics;
-import java.util.ArrayList;
-import java.util.LinkedList;
-
-public class Handler {
-	
-	LinkedList<GameObject> object = new LinkedList<GameObject>();
-	LinkedList<GameObject> toAdd = new LinkedList<GameObject>();
-	
-	public void tick(){
-		// Will declaring this inside of the loop make it clear after each call to tick()?
-		ArrayList<Integer> toRemoveIndex = new ArrayList<Integer>();
-		
-		// Update Objects
-		for(GameObject tempObject : object){
-			tempObject.tick();
-			
-			// Save index of array objects to be removed
-			if (tempObject.removed) {
-				toRemoveIndex.add(object.indexOf(tempObject));
-			}
-		}
-	
-		// Remove marked objects
-		for (int index = toRemoveIndex.size() - 1; index >= 0; index--) {
-			this.removeObject(object.get(toRemoveIndex.get(index)));
-		}
```

Comment:
I was thinking this would just remove the iterator and not the item from the underlying collection, but the documentation for iterator.remove() says differently! Thanks.